lib/faktory_worker/connection.ex

```diff
@@ -100,22 +98,29 @@ defmodule FaktoryWorker.Connection do
 
   defp append_password_hash(args, _, _), do: args
 
-  defp append_worker_fields(args, true) do
-    {:ok, hostname} = :inet.gethostname()
+  defp append_worker_fields(args, opts) do
+    is_worker = Keyword.get(opts, :is_worker, false)
+
+    if is_worker,
+      do: put_worker_args(args, opts),
+      else: args
+  end
+
+  defp put_worker_args(args, opts) do
+    worker_id = Keyword.get(opts, :worker_id)
     sys_pid = System.get_pid()
+    {:ok, hostname} = :inet.gethostname()
 
     worker_args = %{
       hostname: to_string(hostname),
-      wid: Random.worker_id(),
+      wid: worker_id || Random.worker_id(),
```

Comment:
Could make `Random.worker_id()` the fallback value by using the 3rd arg of `get` on line 110?

Reply:
I think I subconsciously did this because this way `Random.worker_id()` doesn't get called when it's not used. If it's specified as the third argument in the `Keyword.get` it will be called to get the value and then conditionally used.